packages/twenty-front/src/modules/object-record/hooks/useCreateManyRecords.ts

```diff
@@ -60,42 +63,83 @@ export const useCreateManyRecords = <
     recordsToCreate: Partial<CreatedObjectRecord>[],
     upsert?: boolean,
   ) => {
-    const sanitizedCreateManyRecordsInput = recordsToCreate.map(
-      (recordToCreate) => {
-        const idForCreation = recordToCreate?.id ?? (upsert ? undefined : v4());
-
-        return {
-          ...sanitizeRecordInput({
-            objectMetadataItem,
-            recordInput: recordToCreate,
-          }),
-          id: idForCreation,
-        };
-      },
-    );
-
-    const recordsCreatedInCache: ObjectRecord[] = [];
+    const { sanitizedCreateManyRecordsInput, recordOptimisticRecordsInput } =
+      recordsToCreate.reduce<
+        Record<
+          'sanitizedCreateManyRecordsInput' | 'recordOptimisticRecordsInput',
+          Partial<ObjectRecord>[]
+        >
+      >(
+        (acc, recordToCreate) => {
```

Comment:
let's avoid reducing as this is more complex for the brain to apprehend

Reply:
Will do using a map and tuple return